handler/src/test/java/io/netty/handler/ssl/SniHandlerTest.java

```diff
@@ -395,6 +403,8 @@ protected void configurePipeline(ChannelHandlerContext ctx, String protocol) {
 
                 assertTrue(serverApnDoneLatch.await(5, TimeUnit.SECONDS));
                 assertTrue(clientApnDoneLatch.await(5, TimeUnit.SECONDS));
+                assertEquals(0, serverApnCtxLatch.getCount());
+                assertEquals(0, clientApnCtxLatch.getCount());
```

Comment:
please use `await(...)` and not `getCount`. if you really not want to call `await(...)` just replace the latch usage with an `AtomicBoolean`.

Reply:
I didn't want to use ```await(...)``` because it would introduce a 5 second delay on each invocation. An ```AtomicBoolean``` solves that problem nicely, thanks.